edx_event_bus_kafka/internal/producer.py

```diff
@@ -157,6 +167,39 @@ def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
     return key_serializer, value_serializer
 
 
+class ProducingContext:
+    """
+    Wrapper class to allow us to link a call to produce() with the on_event_deliver callback
+    """
+    def __init__(self, context_data):
+        self.context_data = context_data
+
+    def __str__(self):
+        return " ".join([f"{key}={value}" for key, value in self.context_data.items()])
+
+    def set_context_data(self, key, value):
+        self.context_data[key] = value
+
+    def on_event_deliver(self, err, evt):
+        """
+        Simple callback method for debugging event production
+
+        If there is any error, log all the known information about the calling context so the event can be recreated
+        and/or replayed later
+
+        Arguments:
+            err: Error if event production failed
+            evt: Event that was delivered (or failed to be delivered)
+        """
+        if err is not None:
+            logger.warning(
+                ERROR_STRING_TEMPLATE.format(error=err, context=self)  # pylint: disable=logging-format-interpolation
+            )
+        else:
+            logger.info(f"Event delivered to topic {evt.topic()}; key={evt.key()}; "
```

Comment:
I imagine this would be too noisy for logs. Maybe debug? We could also wrap it in a toggle, if you think it is important for now.

Reply:
I rather like having this for now. I also don't think it will be all that noisy. If it becomes so, we can add a toggle later.

Reply:
[wontfix unless you want to argue about it]

Reply:
I'm fine leaving this for now. Keep in mind that this is a single event, and I'm hopeful that the Event Bus will replace most server-to-server communication in the future, so I am hoping this gets bad. We'll see.

Temporarily marking unresolved, just so you see this, not for you to take action.